src/bindings/swig/ruby/CMakeLists.txt

```diff
@@ -63,8 +63,11 @@ else()
 	set (SWIG_COMPILE_FLAGS "${SWIG_COMPILE_FLAGS} -DSWIG_TYPE_TABLE=kdb")
 
 	# disable certain compiler warnings for SWIG generated files
-	set (SWIG_COMPILE_FLAGS "${SWIG_COMPILE_FLAGS} -Wno-unused-parameter -Wno-unused-but-set-variable")
+	set (SWIG_COMPILE_FLAGS "${SWIG_COMPILE_FLAGS} -Wno-unused-parameter")
 	set (SWIG_COMPILE_FLAGS "${SWIG_COMPILE_FLAGS} -Wno-sign-compare")
+	if (NOT CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Comment:
Is this Clang or Apple specific?

Reply:
As far as I can tell the compiler warning `unused-but-set-variable` is only supported by GCC. I changed the check to

```cmake
if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
```

.